web/app.js

```diff
@@ -2105,14 +2105,21 @@ const PDFViewerApplication = {
   unbindWindowEvents() {
     this._windowAbortController?.abort();
     this._windowAbortController = null;
-    if (
-      (typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")) ||
-      AppOptions.get("isInAutomation")
-    ) {
-      this._globalAbortController?.abort();
-      this._globalAbortController = null;
-      this.l10n?.pause();
-    }
+  },
+
+  /**
+   * @ignore
+   */
+  async testingClose() {
+    this.l10n?.pause();
+
+    await this.close();
+
+    this.unbindEvents();
+    this.unbindWindowEvents();
```

Comment:
I think it's better to remove the different listeners before closing, because closing can trigger some events.

Reply:
Closing can trigger some events, e.g. related to scripting, that we might want to handle?

Reply:
... or is that not something that we need to worry about here?